skeleton/local-deploy/deployment/README.md

```diff
@@ -0,0 +1,78 @@
+# Deploying Locally
+
+This document is intended for users who wish to deploy their application locally. This was generated from a software template and information regarding image building and deployment are found below. 
+
+## Quick Links
+
+- [Prerequisites](#prerequisite)
+
+- [Building and Pushing Images](#building-and-pushing-image-manually)
+
+- [Deployment](#deploying-to-podman-desktop-ai-lab)
+
+- [Cleanup](#cleanup)
+
+- [Troubleshooting](#troubleshooting)
+
+## Prerequisite
+
+Deploying locally via Podman Desktop is done with the assistance of the `AI Lab Extension`. Please ensure you have the following downloaded and installed:
+- [Podman Desktop 1.8.0+](https://github.com/containers/podman-desktop)
+- [Podman 4.9.0+](https://podman.io/)
+- [Podman Desktop AI Extension](https://github.com/containers/podman-desktop-extension-ai-lab?tab=readme-ov-file#installation)
+
+## Building and Pushing Image Manually
+
+You have chosen `${{ values.imageRegistry }}` as your desired image registry. Please ensure you are logged in and can read/write images from that repository. For example using `quay.io` you can run this command to prompt the entering of credentials:
+```
+podman login quay.io
+```
+
+To rebuild images you can run the following command from the root of the repository:
+```
+bash deployment/build_and_push.sh <absolute path to docker/containerfile>
+```
+
+This script requries you to provide the absolute path to your docker/containerfile as an argument. The script will perform both the building of the image as well as pushing it to the registry location you specified as part of the developer hub setup for this application. You can view the templated values in `build_and_push.sh`.
+
+## Deploying To Podman Desktop AI Lab
+
+**Note: Podman Desktop does not require images to be prebuilt before deployment. If the images are missing it will build them locally for you.**
+
+Deploying via the `AI Lab` extension requires adding the generated `user-catalog.json` file to the Podman Desktops `extensions-storage`. In the [official documentation](https://github.com/containers/podman-desktop-extension-ai-lab?tab=readme-ov-file#-providing-a-custom-catalog) for the extension the default and most common install location is `$HOME/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab`. If you altered the location please take note of the path to your `extensions-storage`. 
```

Comment:
FWIW, I installed the AI Lab extension and `$HOME/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab` is not present on my machine

Reply:
Really? That's interesting since we presumably installed it the same way on our laptops. I wonder if they changed it and haven't updated the docs

Reply:
It may be because I haven't run any of the recipes yet on my new laptop?

Reply:
Potentially, I never thought of that since I tried the recipes once AI Lab released so I didn't think to check

Reply:
Does `$HOME/.local/share/containers/podman-desktop/` exist for you? That would make sense if it generated the rest for you after you used a recipe.

Reply:
Yup, that makes sense.

Reply:
I assume the deploy script fails for your case then? If it does I need to do a dive into why that directory doesn't exist but their docs outlines it as a step

Reply:
Most likely. I created the directory before running the script so can’t say for certain

Reply:
Reached out to the podman desktop team to inquire about why the directory would be missing. In the meantime I've added a note in the readme for steps to take if it is missing

Reply:
Supposedly this issue should be resolved in next release: https://kubernetes.slack.com/archives/C04A0L7LUFM/p1720799547879799?thread_ts=1720629461.190879&cid=C04A0L7LUFM